packages/flutter/lib/src/material/slider_theme.dart

```diff
@@ -0,0 +1,635 @@
+  ///
+  /// Usually these are the primary, dark and light colors from
+  /// a [ThemeData].
+  ///
+  /// The opacities of these colors will be ignored when assigning
+  /// opacities for the theme component colors.
+  ///
+  /// This is used to generate the default slider theme for a [ThemeData].
+  factory SliderThemeData.materialDefaults({
+    @required Color primaryColor,
+    @required Color primaryColorDark,
+    @required Color primaryColorLight,
+  }) {
+    assert(primaryColor != null);
+    assert(primaryColorDark != null);
+    assert(primaryColorLight != null);
+
+    // These are Material Design defaults, and are used to derive
+    // component Colors (with opacity) from base colors.
+    const int activeRailAlpha = 0xff;
+    const int inactiveRailAlpha = 0x3d; // 24% opacity
+    const int disabledActiveRailAlpha = 0x52; // 32% opacity
+    const int disabledInactiveRailAlpha = 0x1f; // 12% opacity
+    const int activeTickMarkAlpha = 0x8a; // 54% opacity
+    const int inactiveTickMarkAlpha = 0x8a; // 54% opacity
+    const int disabledActiveTickMarkAlpha = 0x1f; // 12% opacity
+    const int disabledInactiveTickMarkAlpha = 0x1f; // 12% opacity
+    const int thumbAlpha = 0xff;
+    const int disabledThumbAlpha = 0x52; // 32% opacity
+    const int valueIndicatorAlpha = 0xff;
+
+    // TODO(gspencer): We don't really follow the spec here for overlays.
+    // The spec says to use 16% opacity for drawing over light material,
+    // and 32% for colored material, but we don't really have a way to
+    // know what the underlying color is, so there's no easy way to
+    // implement this. Choosing the "light" version for now.
+    const int overlayLightAlpha = 0x29; // 16% opacity
+
+    return new SliderThemeData(
+      activeRailColor: primaryColor.withAlpha(activeRailAlpha),
+      inactiveRailColor: primaryColor.withAlpha(inactiveRailAlpha),
+      disabledActiveRailColor: primaryColorDark.withAlpha(disabledActiveRailAlpha),
+      disabledInactiveRailColor: primaryColorDark.withAlpha(disabledInactiveRailAlpha),
+      activeTickMarkColor: primaryColorLight.withAlpha(activeTickMarkAlpha),
+      inactiveTickMarkColor: primaryColor.withAlpha(inactiveTickMarkAlpha),
+      disabledActiveTickMarkColor: primaryColorLight.withAlpha(disabledActiveTickMarkAlpha),
+      disabledInactiveTickMarkColor: primaryColorDark.withAlpha(disabledInactiveTickMarkAlpha),
+      thumbColor: primaryColor.withAlpha(thumbAlpha),
+      disabledThumbColor: primaryColorDark.withAlpha(disabledThumbAlpha),
+      overlayColor: primaryColor.withAlpha(overlayLightAlpha),
+      valueIndicatorColor: primaryColor.withAlpha(valueIndicatorAlpha),
+      thumbShape: const DefaultSliderThumbShape(),
+      valueIndicatorShape: const DefaultSliderValueIndicatorShape(),
+      showValueIndicator: ShowValueIndicator.onlyForDiscrete,
+    );
+  }
+
+  /// A slider theme created from the given theme, with Material Design defaults for
+  /// opacity and text style.
```

Comment:
I would assume that this theme would typically be the same as `Theme.of(context).sliderTheme`?  I guess I'm not sure why this factory constructor is needed.

Reply:
Yeah, this is really just a convenience if you have a Theme already.  I've removed it.